sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/QuotingUtils.scala

```diff
@@ -45,12 +45,12 @@ object QuotingUtils {
 
   private val validIdentPattern = Pattern.compile("^[a-zA-Z_][a-zA-Z0-9_]*")
 
+  def needQuote(part: String): Boolean = {
+    !validIdentPattern.matcher(part).matches()
+  }
+
   def quoteIfNeeded(part: String): String = {
-    if (validIdentPattern.matcher(part).matches()) {
-      part
-    } else {
-      quoteIdentifier(part)
-    }
+    if (needQuote(part)) quoteIdentifier(part) else part
```

Comment:
At the first glance, there is no logical change. Do we need to touch this file?

Reply:
Changes here are piggyback stuff for extracting the `needQuote` method for https://github.com/apache/spark/pull/49506/files#diff-81dcb1eea508dd5886eadedce48c89169b812224020a75d38e66aed0ee71a039R1450

Reply:
Got it. The reuse makes sense to me.